test/error_test.exs

```diff
@@ -0,0 +1,143 @@
+
+      assert err.error_context == ["some context"]
+    end
+
+    test "has a context field populated when there is a list of errors" do
+      test_error1 = TestError.exception(some_field: :a)
+      test_error2 = TestError.exception(some_field: :b)
+
+      err = Ash.Error.to_error_class([test_error1, test_error2], error_context: "some context")
+
+      assert err.error_context == ["some context"]
+    end
+
+    test "accumulates error_context field in child errors" do
+      error1 = Ash.Error.to_ash_error("whoops!", nil, error_context: "some context")
+      error2 = Ash.Error.to_ash_error("whoops, again!!", nil, error_context: "some other context")
+
+      error_class =
+        Ash.Error.to_error_class([error1, error2], error_context: "some higher context")
+
+      child_error_1 = Enum.find(error_class.errors, fn err -> err.error == "whoops!" end)
+      assert child_error_1.error_context == ["some higher context", "some context"]
+
+      child_error_2 = Enum.find(error_class.errors, fn err -> err.error == "whoops, again!!" end)
+      assert child_error_2.error_context == ["some higher context", "some other context"]
+    end
+
+    test "accumulates error_context field in child errors who have no error_context of their own" do
+      error1 = Ash.Error.to_ash_error("whoops!", nil, error_context: "some context")
+      error2 = Ash.Error.to_ash_error("whoops, again!!", nil)
+
+      error_class =
+        Ash.Error.to_error_class([error1, error2], error_context: "some higher context")
+
+      child_error_1 = Enum.find(error_class.errors, fn err -> err.error == "whoops!" end)
+      assert child_error_1.error_context == ["some higher context", "some context"]
+
+      child_error_2 = Enum.find(error_class.errors, fn err -> err.error == "whoops, again!!" end)
+      assert child_error_2.error_context == ["some higher context"]
+    end
+
+    test "leaves child error contexts unchanged if no error_context field provided" do
+      error1 = Ash.Error.to_ash_error("whoops!", nil, error_context: "some context")
+      error2 = Ash.Error.to_ash_error("whoops, again!!", nil, error_context: "some other context")
+
+      error_class = Ash.Error.to_error_class([error1, error2])
+
+      child_error_1 = Enum.find(error_class.errors, fn err -> err.error == "whoops!" end)
+      assert child_error_1.error_context == ["some context"]
+
+      child_error_2 = Enum.find(error_class.errors, fn err -> err.error == "whoops, again!!" end)
+      assert child_error_2.error_context == ["some other context"]
+    end
+
+    test "error message contains error context breadcrumbs" do
+      error1 = Ash.Error.to_ash_error("whoops!", nil, error_context: "some context")
+      error2 = Ash.Error.to_ash_error("whoops, again!!", nil, error_context: "some other context")
+
+      error_class =
+        Ash.Error.to_error_class([error1, error2], error_context: "some higher context")
+
+      error_message = Ash.Error.Unknown.message(error_class)
```

Comment:
The error message value for this error is something like:
```Elixir
Unknown Error

Context: some higher context > some context
* whoops!
  (elixir 1.13.4) lib/process.ex:773: Process.info/2
  (ash 2.2.0) lib/ash/error/unknown/unknown_error.ex:5: Ash.Error.Unknown.UnknownError.exception/1
  (ash 2.2.0) lib/ash/error/error.ex:147: Ash.Error.to_ash_error/3
  test/error_test.exs:117: Ash.Test.ErrorTest."test to_error_class error message"/1
  (ex_unit 1.13.4) lib/ex_unit/runner.ex:500: ExUnit.Runner.exec_test/1
  (stdlib 4.1) timer.erl:235: :timer.tc/1
  (ex_unit 1.13.4) lib/ex_unit/runner.ex:451: anonymous fn/4 in ExUnit.Runner.spawn_test_monitor/4
Context: some higher context > some other context
* whoops, again!!
  (elixir 1.13.4) lib/process.ex:773: Process.info/2
  (ash 2.2.0) lib/ash/error/unknown/unknown_error.ex:5: Ash.Error.Unknown.UnknownError.exception/1
  (ash 2.2.0) lib/ash/error/error.ex:147: Ash.Error.to_ash_error/3
  test/error_test.exs:118: Ash.Test.ErrorTest."test to_error_class error message"/1
  (ex_unit 1.13.4) lib/ex_unit/runner.ex:500: ExUnit.Runner.exec_test/1
  (stdlib 4.1) timer.erl:235: :timer.tc/1
  (ex_unit 1.13.4) lib/ex_unit/runner.ex:451: anonymous fn/4 in ExUnit.Runner.spawn_test_monitor/4
```

Reply:
Almost the same as your suggestion in the issue.  I can tweak this more if need be.

Reply:
looks great!